File: mario-less/mario.c
// Prompts user for an int height and creates a Mario-style half-pyramid with # characters

#include <stdio.h>
#include <cs50.h>

// Continually prompts user for int height until positive
int get_height(void);

// Prints half-pyramid with size input
void print_hpyramid(int size);


int main(void)
{
    // Get height of half-pyramid
    int h = get_height();

    // Print half-pyramid
    print_hpyramid(h);
}










// Continually prompts user for int height until positive
int get_height(void)
{
    int h;
    do
    {
        h = get_int("Height: ");
    }
    while (h < 1);
    return h
}

// Prints half-pyramid with height input
void print_hpyramid(int height)
{
    // Print input number of rows
    while (i = 0, i < height, i++)
    {
        // In each row, print decreasing spaces starting with height - 1
        while (j = 0, j < height - 1 - i, j++)
        {
            print(" ");
        }

        // Then print increasing # followed by \n starting with 1
        while (k = 0, 

        // printf()
        // print size-1_ 1"#" "\n"
        // print size-2_ 2# "\n"
        // print size-3_ 3# "\n"
        // print size-size_ size# "\n"
    }


}

File: cs50/week6/pset/sentimental-mario-less/mario.py
"""
Prompts user for an int height and creates a Mario-style half-pyramid
with # characters.
"""


def main():
    """Take user input and print pyramid."""

    # Get height of half-pyramid
    height = get_height()

    # Print half-pyramid
    print_hpyramid(height)


def get_height() -> int:
    """Return user-input positive integer pyramid height."""

    # Continually re-prompt if height is not between 1 and 8, inclusive.
    while True:
        h = int(input('Height: '))
        if 1 <= h and h <= 9:
            break

        # Prompt user to reenter height.
        print('error: height must be positive integer')

    return h


def print_hpyramid(height: int):
    """Print half-pyramid with height input."""

    # Print input number of rows.
    for row in range(height):

        # In each row, print decreasing spaces starting with height - 1.
        for space in range(height - 1 - row):
            print(' ', end='')

        # Then print increasing # starting with 1.
        for block in range(row + 1):
            print('#', end='')

        # Print new line after each row
        print()


if __name__ == "__main__":
    main()
